backend/src/index.ts

```diff
@@ -317,11 +329,16 @@ app.get(
     const pageCount = Math.ceil(size / increments);
 
     const data: ScoreEntry[] = [];
-
-    for (let i = 0; i < increments && i + start < size; i++) {
+    const end = Math.min(start + increments, size);
+    for (let i = start; i < end; i++) {
+      // if username is undefined, don't add to leaderboard
+      let username: string;
+      if (!(username = await getUsername(queryScoreSnapshot.docs[i + start].data().userid))) {
```

Comment:
shouldn't this throw an error or something

Reply:
something like  `throw error("Invalid userid in games")` ?